maat-court-data-api/src/main/java/gov/uk/courtdata/reporderhistory/entity/RepOrderHistoryEntity.java

```diff
@@ -0,0 +1,240 @@
+package gov.uk.courtdata.reporderhistory.entity;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import gov.uk.courtdata.entity.RepOrderEntity;
+import jakarta.persistence.*;
+import lombok.*;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "REP_ORDERS_HISTORY", schema = "TOGDATA")
+public class RepOrderHistoryEntity {
+
+    @ToString.Exclude
+    @JsonBackReference
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "REP_ID", nullable = false, updatable = false)
+    private RepOrderEntity repOrder;
+
+    @Id
+    @Column(name = "ID")
+    private Integer id;
+
+    @Column(name = "CASE_ID")
+    private String caseId;
```

Comment:
I had a quick look in the dev DB on this table and there are constraints such as this field being not null on the table. It is usually helpful to catch these type of errors by having the constraints here in the entity before the DB throwing them.

Reply:
I've not added them to this particular entity because I only created this entity for the purposes of cascade deleting when deleting a rep order (I need this new RepOrderHistoryEntity to do the delete). We don't do any creation or updating of the rep_orders_history table within this application - this all happens in DB triggers within stored procedures in MAAT, so don't need the validation on the entity at the moment.

However, I have now added @NotNull annotations to the required fields on my CreateRepOrder class, so they are now caught as a BadRequest if not submitted with the POST request.